base/show.jl

```diff
@@ -1442,20 +1505,24 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int)
     elseif head === :where && nargs > 1
         parens = 1 <= prec
         parens && print(io, "(")
-        show_unquoted(io, args[1], indent, operator_precedence(:(::)))
+        show_unquoted(io, args[1], indent, operator_precedence(:(::)), quote_level)
         print(io, " where ")
         if nargs == 2
-            show_unquoted(io, args[2], indent, 1)
+            show_unquoted(io, args[2], indent, 1, quote_level)
         else
             print(io, "{")
-            show_list(io, args[2:end], ", ", indent)
+            show_list(io, args[2:end], ", ", indent, 0, quote_level)
             print(io, "}")
         end
         parens && print(io, ")")
 
-    elseif (head === :import || head === :using) && nargs == 1 &&
-            (valid_import_path(args[1]) ||
-             (Meta.isexpr(args[1], :(:)) && length(args[1].args) > 1 && all(valid_import_path, args[1].args)))
+    elseif (head === :import || head === :using) &&
+           ((nargs == 1 && (valid_import_path(args[1]) ||
+                           (Meta.isexpr(args[1], :(:)) &&
+                            length(args[1].args) > 1 &&
+                            all(valid_import_path, args[1].args)))) ||
+            (all([Meta.isexpr(args[i], :(.)) for i in 1:nargs]) &&
```

Comment:
Doesn't `valid_import_path` already check this?

Reply:
In fact it does.